packages/sdk-core/src/BacktraceCoreApi.ts

```diff
@@ -0,0 +1,97 @@
+import { BacktraceAttachment } from './model/attachment/BacktraceAttachment.js';
+import {
+    BacktraceAttachmentResponse,
+    BacktraceReportSubmissionResult,
+    BacktraceRequestHandler,
+    BacktraceSubmitBody,
+    BacktraceSubmitResponse,
+    RequestBacktraceReportSubmission,
+} from './model/http/index.js';
+import {
+    BacktraceSubmitSummedMetricsBody,
+    BacktraceSubmitUniqueMetricsBody,
+} from './model/http/model/metric/request/BacktraceSubmitMetricsBody.js';
+import { MetricsUrlInformation } from './modules/metrics/MetricsUrlInformation.js';
+
+export interface BacktraceCoreApiOptions {
+    readonly url: string;
+    readonly token?: string;
+
+    readonly metrics?: {
+        readonly url?: string;
+    };
+
+    readonly requestHandler?: BacktraceRequestHandler;
+    readonly requestBacktraceReportSubmission?: RequestBacktraceReportSubmission;
+}
+
+export class BacktraceCoreApi {
+    private readonly _summedMetricsSubmissionUrl?: string;
+    private readonly _uniqueMetricsSubmissionUrl?: string;
+
+    private readonly _requestHandler: BacktraceRequestHandler;
+    private readonly _requestBacktraceReportSubmission: RequestBacktraceReportSubmission;
+
+    constructor(options: BacktraceCoreApiOptions, requestHandler: BacktraceRequestHandler) {
+        this._summedMetricsSubmissionUrl = MetricsUrlInformation.generateSummedEventsUrl(
+            options.metrics?.url ?? 'https://events.backtrace.io',
+            options.url,
+            options.token,
+        );
+
+        this._uniqueMetricsSubmissionUrl = MetricsUrlInformation.generateUniqueEventsUrl(
+            options.metrics?.url ?? 'https://events.backtrace.io',
+            options.url,
+            options.token,
+        );
+
+        this._requestHandler = options.requestHandler ?? requestHandler;
```

Comment:
requestHandler is always defined. It's weird that you want to override it - that's why you pass it as a second parameter, and you still use the one from the options. Maybe the client's backtrace API should override options if the request handler is not defined?

Reply:
Removed requestHandler from core options.